tests/system/common.py

```diff
@@ -819,3 +823,54 @@ def kill_process_on_host(hostname, pattern):
     else:
         print("Killed no pids")
     return pids
+
+
+@lru_cache()
+def dcos_masters_public_ips():
+    @retrying.retry(
+        wait_fixed=1000,
+        stop_max_attempt_number=240,  # waiting 20 minutes for exhibitor start-up
+        retry_on_exception=ignore_provided_exception(DCOSException))
+    def all_master_ips():
+        return get_all_master_ips()
+
+    master_private_ips = all_master_ips()
+
+    master_public_ips = []
+
+    for private_ip in master_private_ips:
```

Comment:
nit: you could just `map` over the private IPs e.g.: `public_ips = list(map(shakedown.run_command(...)[1], private_ips)`

Reply:
it's not a very python-ish style :)

Reply:
Well, the following is more pythonic 🙂 

```python
master_public_ips = [shakedown.run_command(private_ip, '/opt/mesosphere/bin/detect_ip_public')[1]
                     for private_ip in master_private_ips]
```

this swallows `results` though.

Also, why not

```python
for private_ip in all_master_ips():
  ...
```